src/Providers/Http/DTO/Request.php

```diff
@@ -0,0 +1,383 @@
+     */
+    protected array $headers;
+
+    /**
+     * @var array<string, string> Map of lowercase header names to actual header names for fast lookup.
+     */
+    protected array $headersMap;
+
+    /**
+     * @var string|array<string, mixed>|null The request data.
+     */
+    protected $data;
+
+    /**
+     * Constructor.
+     *
+     * @since n.e.x.t
+     *
+     * @param HttpMethodEnum $method The HTTP method.
+     * @param string $uri The request URI.
+     * @param array<string, string|list<string>> $headers The request headers.
+     * @param string|array<string, mixed>|null $data The request data.
+     *
+     * @throws InvalidArgumentException If the URI is empty.
+     */
+    public function __construct(HttpMethodEnum $method, string $uri, array $headers = [], $data = null)
+    {
+        if (empty($uri)) {
+            throw new InvalidArgumentException('URI cannot be empty.');
+        }
+
+        $this->method = $method;
+        $this->uri = $uri;
+        $this->headers = $this->normalizeHeaders($headers);
+        $this->headersMap = $this->buildHeadersMap($this->headers);
+        $this->data = $data;
+    }
+
+    /**
+     * Gets the HTTP method.
+     *
+     * @since n.e.x.t
+     *
+     * @return HttpMethodEnum The HTTP method.
+     */
+    public function getMethod(): HttpMethodEnum
+    {
+        return $this->method;
+    }
+
+    /**
+     * Gets the request URI.
+     *
+     * For GET requests with array data, appends the data as query parameters.
+     *
+     * @since n.e.x.t
+     *
+     * @return string The URI.
+     */
+    public function getUri(): string
+    {
+        // If GET request with array data, append as query parameters
+        if ($this->method === HttpMethodEnum::GET() && is_array($this->data) && !empty($this->data)) {
+            $separator = strpos($this->uri, '?') === false ? '?' : '&';
+            return $this->uri . $separator . http_build_query($this->data);
+        }
+
+        return $this->uri;
+    }
+
+    /**
+     * Gets the request headers.
+     *
+     * @since n.e.x.t
+     *
+     * @return array<string, list<string>> The headers.
+     */
+    public function getHeaders(): array
+    {
+        return $this->headers;
+    }
+
+    /**
+     * Gets a specific header value.
+     *
+     * @since n.e.x.t
+     *
+     * @param string $name The header name (case-insensitive).
+     * @return list<string>|null The header value(s) or null if not found.
+     */
+    public function getHeader(string $name): ?array
+    {
+        $lower = strtolower($name);
+        if (!isset($this->headersMap[$lower])) {
+            return null;
+        }
+        return $this->headers[$this->headersMap[$lower]];
+    }
+
+    /**
+     * Gets the first value of a specific header.
+     *
+     * @since n.e.x.t
+     *
+     * @param string $name The header name (case-insensitive).
+     * @return string|null The first header value or null if not found.
```

Comment:
This isn't correct, it returns all header parts concatenated.

Reply:
Resolved in a632982